tests/test_trial.py

```diff
@@ -1319,3 +1319,25 @@ def objective(trial: optuna.trial.Trial) -> Tuple[float, float]:
         )
 
     study.optimize(objective, n_trials=10)
+
+
+def test_report_with_multi_objectives() -> None:
+    study = optuna.create_study(directions=["maximize", "maximize"])
+    trial = Trial(study, study._storage.create_new_trial(study._study_id))
```

Comment:
Let's prefer public APIs when possible, i.e. using `Study.optimize` to construct trials.